File: twoSum.py
'''

Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

'''


def twoSum(nums, target):
    mydict = {}

    for k in range(0, len(nums)):

        if target - nums[k] in mydict.keys():
            return [mydict[target - nums[k]], k]
        else:
            mydict[nums[k]] = k

    return 0

print(twoSum([2,7,8,11],9))

File: twosum.java
/**
Given an array of integers, return indices of the two numbers such that they add up to a specific target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.
*/

class Solution {
    public int[] twoSum(int[] nums, int target) {

        if(nums.length < 2)
            return null;

        int[] res = new int[2];
        Map<Integer,Integer> mp = new HashMap<Integer, Integer>();

        for(int i=0; i< nums.length; i++)
        {
            Integer diff = (Integer) (target - nums[i]);
            if(mp.containsKey(diff))
            {
                res[0] = mp.get(diff);
                res[1] = i;
                return res;
            }
            else
            {
                mp.put(nums[i], i);
            }

        }
        return null;

    }
}